### PR TITLE
feat: Install uloop natively on macOS

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- macOS installs can now finish native PATH setup from the downloaded uloop binary, matching the Windows native handoff flow.
- The installer keeps legacy npm cleanup and fallback guidance reliable when native setup is unavailable or needs manual repair.

## User Impact
- macOS users can install with the shell installer and get a persistent uloop PATH entry for zsh, bash, or fish.
- Legacy npm uloop-cli shims are detected and cleaned up when possible, with manual guidance when automatic cleanup cannot safely finish.
- Older release assets still fall back to the compatibility installer path instead of being treated as native-install capable.

## Changes
- Add macOS support to the native install command, including profile block updates, symlinked profile handling, path quoting, and PATH shadowing reports.
- Route scripts/install.sh through native setup when the staged binary advertises current macOS or Windows install support, with compatibility fallback on failure.
- Add regression coverage for shell profile updates, legacy npm cleanup edge cases, release asset probing, and checked-in native CLI binaries.

## Verification
- go test ./internal/cli ./internal/install -count=1
- scripts/check-go-cli.sh
- scripts/test-install-release-filter.sh
- codex-review v3-beta with parallel tests: clean, no accepted/actionable findings reported

### CODE REVIEW

File: Packages/src/Cli~/internal/install/posix_install.go [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+		posixInstallScript(installDir, targetPath),
+	}
+}
+
+func posixInstallScript(installDir string, targetPath string) string {
+	return fmt.Sprintf(
+		`InstallDir=%s
+ExpectedUloopPath=%s
+PathBlockStart="# >>> uloop PATH >>>"
+PathBlockEnd="# <<< uloop PATH <<<"
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/install/posix_install.go:20]: Thanks for checking this. I verified the current code and I don't think this is a valid issue.

The `__ULOOP_PATH_START__` / `__ULOOP_PATH_END__` markers in `CliInstallationDetector` are stdout delimiters for the shell detection command that runs `command -v uloop`. They are not the shell-profile block markers written by the installer.

The macOS native installer's `# >>> uloop PATH >>>` / `# <<< uloop PATH <<<` block is used only by the installer's own profile replacement logic in `posix_install.go`, so it does not need to match the detector markers. No code change was made for this thread.

File: Packages/src/Cli~/internal/install/posix_install.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+        /*)
+            echo "$link_target"
+            ;;
+        *)
+            profile_dir=${profile_path%%/*}
+            if [ "$profile_dir" = "$profile_path" ]; then
+                echo "$link_target"
+                return
+            fi
+            echo "$profile_dir/$link_target"
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/install/posix_install.go:118]: Thanks. I verified this against the generated shell script and I don't think this is a valid issue.

The `posixInstallScript` source is a Go `fmt.Sprintf` template, so `%%` in the Go source becomes a single `%` in the runtime shell script. The executed script therefore uses dirname-style `${profile_path%/*}`, `${profile_write_path%/*}`, and `${command_path%/*}` expansions, not `${...%%/*}`.

I also reran the focused install tests covering nested fish profile creation, symlinked profiles, absolute npm shim cleanup, and manual cleanup guidance; they pass. No code change was made for this thread.

File: Packages/src/Cli~/internal/install/posix_install.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+            ;;
+    esac
+}
+
+write_path_block() {
+    profile_path=$1
+    path_line=$2
+    profile_write_path=$(resolve_profile_write_path "$profile_path")
+    profile_dir=${profile_write_path%%/*}
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/install/posix_install.go:127]: Thanks. I verified this against the generated shell script and this does not reproduce in the runtime script.

The Go source contains `${profile_write_path%%/*}` because it is inside a `fmt.Sprintf` template. At runtime that becomes `${profile_write_path%/*}`, which preserves the parent directory for absolute paths such as `/tmp/home/.config/fish/config.fish`. The existing focused fish profile test also passes. No code change was made for this thread.

File: Packages/src/Cli~/internal/install/posix_install.go [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+        echo "$profile_path"
+        return
+    fi
+
+    case "$link_target" in
+        /*)
+            echo "$link_target"
+            ;;
+        *)
+            profile_dir=${profile_path%%/*}
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/install/posix_install.go:113]: Thanks. I checked this against the generated shell script and this is not a runtime bug.

The Go source uses `%%` to escape a literal percent sign through `fmt.Sprintf`. The shell receives `${profile_path%/*}`, so relative symlink targets are resolved against the actual parent directory for absolute profile paths. The focused symlinked profile test passes. No code change was made for this thread.

File: Packages/src/Cli~/internal/install/posix_install.go [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+            return 1
+            ;;
+    esac
+}
+
+infer_npm_prefix_from_uloop_path() {
+    command_path=$1
+    case "$command_path" in
+        */bin/uloop|*/bin/uloop.exe)
+            bin_dir=${command_path%%/*}
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/install/posix_install.go:201]: Thanks. I verified this against the generated shell script and the npm prefix inference is using the intended runtime expansion.

The Go source has `${command_path%%/*}` only because `posixInstallScript` is formatted through `fmt.Sprintf`. The shell receives `${command_path%/*}`, which keeps the absolute `.../bin` parent and then strips `/bin` for the prefix. The focused absolute legacy npm shim cleanup tests pass. No code change was made for this thread.
